packages/terminal/src/node/shell-process.ts

```diff
@@ -20,9 +20,23 @@ export type ShellProcessFactory = (options: ShellProcessOptions) => ShellProcess
 export const ShellProcessOptions = Symbol("ShellProcessOptions");
 export interface ShellProcessOptions {
     shell?: string,
+    args?: string[],
     rootURI?: string,
     cols?: number,
-    rows?: number
+    rows?: number,
+    env?: { [key: string]: string | null },
+}
+
+function setUpEnvVariables(optionsEnv?:  { [key: string]: string | null }): NodeJS.ProcessEnv {
+    const prEnv: NodeJS.ProcessEnv = {...process.env};
+    if (optionsEnv) {
+        for (const envName of Object.keys(optionsEnv)) {
+            const envValue = optionsEnv[envName];
+            prEnv[envName] = envValue || '';
```

Comment:
Strange, that you can have a value `string | null` but eventually, `null` values will be ignored and substituted with `''` (empty strings) under the hood.

Reply:
@kittaakos, thanks. I digged deeper. For creation terminal process we are using node-pty lib. With help this lib we spawn terminal process:
https://github.com/theia-ide/theia/blob/e7eb69fc125ddcc75ca413816f33b2d1272d4c5c/packages/process/src/node/terminal-process.ts#L41-L44
Inside this lib used interface with process options definition IPtyForkOptions (for us it's `options.options`, but we defined type like `object`): 
```
export interface IPtyForkOptions {
  name?: string;
  cols?: number;
  rows?: number;
  cwd?: string;
  env?: ProcessEnv;
  uid?: number;
  gid?: number;
  encoding?: string;
}
```
Where is ProcessEnv:
```
    export type ProcessEnv = {
      [key: string]: string
    };
```

But we got env. variables from node, NodeJS.ProcessEnv:
```
    export interface ProcessEnv {
        [key: string]: string | undefined;
    }
```
So it's normal if we replace not initialized value to the empty value, for compatibility interfaces. But, yes, we should do it for all envs variables. So I will improve it.